### PR TITLE
PAL: Fix ptrace-invocation on FreeBSD.

### PR DESCRIPTION
**Pull request by josteink**
Unlike in Linux[1] FreeBSD's ptrace does not use an enum[2].

[1] http://linux.die.net/include/sys/ptrace.h
[2] http://www.freebsd.org/cgi/man.cgi?query=ptrace

This fixes compiler errors like this one:

    /home/josteink/build/coreclr/src/pal/src/arch/i386/context.cpp:305:21: error: use of undeclared identifier '__ptrace_request'
            if (ptrace((__ptrace_request)PT_GETREGS, processId, (caddr_t) &ptrace_registers, 0) == -1)

### CODE REVIEW

File: src/pal/src/arch/i386/context.cpp [hunk 1]
```diff
@@ -298,15 +298,20 @@ BOOL CONTEXT_GetRegisters(DWORD processId, ucontext_t *registers)
     {
 #if HAVE_PT_REGS
         struct pt_regs ptrace_registers;
+        if (ptrace((__ptrace_request)PT_GETREGS, processId, (caddr_t) &ptrace_registers, 0) == -1)
+	{
```
Comment (jkotas) [inline src/pal/src/arch/i386/context.cpp]: The indentation is one space less than what it should be.

**Comment by jkotas** [comment]
LGTM otherwise (could you please squash after fixing the cosmetic comment)

**Comment by josteink** [comment]
Done and done.

**Comment by kangaroo** [comment]
One nit comment.  In the top piece you duplicate the assert between #defines, and in the latter two you have a single assert and #if the usage.  It would be nice to unify them to be consistent.

**Comment by josteink** [comment]
Looking at it again, I would have to agree. Let me get that fixed.

**Comment by jkotas** [comment]
@josteink, @janhenke, @kangaroo and the FreeBSD port team - thanks a lot for steady stream of great PRs!

**Comment by TimLovellSmith** [comment]
Was the explicit cast in the original code ((__ptrace_request)PT_GETREGS) instead of PT_GETREGS actually necessary? Isn't linux header already defining it as an enum value which should have correct type?

**Comment by josteink** [comment]
@TimLovellSmith I honestly don't know, but I didn't want to change existing code for platforms I'm not building, so I decided to go with the ifdef-route.

@jkotas You're welcome! Thanks for accepting them, and quickly! I've contributed to other projects where it's taken weeks to get a patch submitted, and it really kills any sense of momentum and initiative. You're doing a good job here :)

**Pull request merged**